### ISSUE TITLE
General improvements discussion

### ISSUE THREAD

**Issue by fregante** [hankxdev/one-click-extensions-manager#102]
I'm considering some general improvements that are related to each other:

1. Add an options page
2. Let the user pick the "browser_action" behavior
  - extension popup (current)
  - actual popup window (chrome.windows.create) which can stay open and be resized and repositioned
  - tab, which would replace the current "wide view" in the options

I can't think of many other options I want to add though, I still think this extension should be simple. Especially because I don't want to spend much time tweaking and fixing it. 

This is more of a way to "make more sense" since we're using the Options page in a weird way.

**Comment by hankxdev** [comment]
I agree that providing users with options on the options page is a good idea. However, I don't believe that the behavior of the browser_action needs to be changed. It serves as a simple list of extensions with their corresponding actions, and I don't see any compelling reason for users to open it in a new window or tab.

**Comment by fregante** [comment]
> I don't see any compelling reason for users to open it in a new window or tab.

The popup closes every time you click outside. It's the reason why we started showing the extension in `options` as well:

- https://github.com/hankxdev/one-click-extensions-manager/pull/4

And this too:

- https://github.com/hankxdev/one-click-extensions-manager/issues/31

If Firefox ever supports this, it could also be shown in the sidebar: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/sidebar_action#browser_compatibility

**Comment by hankxdev** [comment]
Yes, I am aware of it.

however, in the majority of cases (perhaps 99%?) you just open the popup, disable/enable one or more extensions, and then return to your tabs, correct? I don't see why the popup should remain open.

but giving users more options is a good thing, even 1% of the users need it, it's 400 users :D (yeah we have almost 40k users ). so yeah, if you want to add it, I'm also OK with it. just please keep the popup as the default option.

**Comment by fregante** [comment]
> please keep the popup as the default option.

Yeah definitely, the options won't change the default behavior, which I still think is best. 👍

**Comment by hankxdev** [comment]
<img width="303" alt="image" src="https://user-images.githubusercontent.com/3389447/227778649-b591bbbc-4e61-4393-a311-ce5abe34ec96.png">
<img width="306" alt="image" src="https://user-images.githubusercontent.com/3389447/227778679-3e6201fa-3b6f-4b1b-be45-69b8168b57a1.png">

@fregante the links in the options page are not working. I think it requests either `_blank` or use `chrome.tabs.create` to make the links open in new tab

**Comment by fregante** [comment]
Will send a PR and release now

**Comment by fregante** [comment]
23.3.26 released. Thanks for the report

**Comment by jvthomas90** [comment]
I realize that `23.3.26` has already been released for a little under a week now, but I'd like to request for a minor reversion in behaviour 🙏🏽

For the record, while I haven't touched any of the newly introduced options and left them at their defaults, I do appreciate their introduction as I think it'd beneficial for many users! 👍🏽 That said... that prior "permanent view" is now completely inaccessible 💔 Indeed it seems that that was why #4 was introduced in the first place!

So, would it be possible to re-introduce the permanent view in some other context?
- Perhaps, as was mentioned in the original proposal, the **"MORE..."** button doesn't seem too useful, so perhaps it could be replaced with this button that creates a separate popup window
- Or perhaps this extra button can be placed somewhere in the newly introduced options html? And clicking it opens this permanent view in a new tab or popup window etc?

Basically, I'd like to request for the return of a more permanent view in some form or other than the default ephemeral extension popup which automatically closes whenever focus is lost. _Of all the extension managers I've tried over time, this is the only one that offers a properly implemented and genuine "undo/redo" functionality affording the user fine-grained controls. But as soon as the extension popup closes, that "safety buffer" is essentially lost._

And sure, I might be in the 1% for using it that way as Hank put it, but that's honestly cuz well over 99.999% of other extension managers can't do what this one does in that regard 😅 which is why I chose to loyally keep using this one all this time 😤 the 1% of the time that I do invoke the permanent view because I needed to make extensive (and experimental) new changes to my setup, this one outperformed all the rest like a freaking champion! 🏅 So... please re-introduce the perma-view? 🥺 
- No need to change the default popup behaviour, I use the ephemeral view for quick adjustments like 99% of the time too.
- And I'm definitely not going to say something silly like "take away my options" haha, if anything thanks for continuing to iterate and improve on this project :) 

I just... I think it's a shame that anytime I happen to sneeze and accidentally misclick elsewhere on my screen the extension disappears and I can no longer undo/redo my actions anymore 🤧👀😭 Having the functional ability to undo/redo operations, and then tying it exclusively to an ephemeral view seems somewhat silly at best and wasteful at worst >_<

> **P.S.** I'm not super savvy when it comes to code, so I wont pretend to know whether what I'm asking for is a monumental milestone undertaking or a minor matter of duplicating the existing extension popup onto a separate, permanent view ¯\\\_(ツ)_/¯ So I'll just say thank you for any and all re-consideration you'll give to bringing back the permanent view in some form or another, and if it simply can't be done then that's that 🤝

**Comment by fregante** [comment]
Will do today 👌
